services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -233,43 +213,32 @@ public void testSaveBulkSubmission() throws IOException {
      * @throws IOException if there is any issue
      */
     @Test
-    public void testSubmissionWithMultipleRequests() throws IOException {
-        final String rawXml = Utilities.getRawXml("src/unit-test/resources/", "testXMLValid3.xml");
+    void testSubmissionWithMultipleRequests() throws IOException {
+        final String rawXml = Utilities.getRawXml(PATH_TEST_RESOURCES, "testXMLValid3.xml");
         SdtContext.getContext().setRawInXml(rawXml);
 
         // Activate Mock Generic Dao
         final IBulkSubmission bulkSubmission = this.createBulkSubmission();
         addValidIndividualRequest(bulkSubmission, "ICustReq124");
         addValidIndividualRequest(bulkSubmission, "ICustReq125");
 
-        LOGGER.debug("Size of Individual Requests in Bulk Submission is " +
-                bulkSubmission.getIndividualRequests().size());
-
         mockGenericDao.persist(bulkSubmission);
-        EasyMock.expectLastCall();
 
         // Mock the serviceRequest fetch
         final IServiceRequest serviceRequest = new ServiceRequest();
-        EasyMock.expect(mockGenericDao.fetch(IServiceRequest.class, 1)).andReturn(serviceRequest);
+        when(mockGenericDao.fetch(IServiceRequest.class, 1)).thenReturn(serviceRequest);
 
-        // Replay the EasyMock
-        EasyMock.replay(mockGenericDao);
-
-        EasyMock.expect(mockBulkCustomerDao.getBulkCustomerBySdtId(10)).andReturn(bulkSubmission.getBulkCustomer());
-
-        EasyMock.replay(mockBulkCustomerDao);
+        when(mockBulkCustomerDao.getBulkCustomerBySdtId(10)).thenReturn(bulkSubmission.getBulkCustomer());
 
         final String key =
                 bulkSubmission.getBulkCustomer().getSdtCustomerId() + bulkSubmission.getCustomerReference();
 
         // Setup concurrency map as if validator had done it.
         IInFlightMessage inFlightMessage = new InFlightMessage();
-        inFlightMessage.setCompetingThreads(new HashMap<Thread, Thread>());
+        inFlightMessage.setCompetingThreads(new HashMap<>());
         inFlightMessage.getCompetingThreads().put(Thread.currentThread(), Thread.currentThread());
 
-        EasyMock.expect(mockConcurrencyMap.get(key)).andReturn(inFlightMessage);
-        EasyMock.expect(mockConcurrencyMap.put(key, inFlightMessage)).andReturn(inFlightMessage);
-        EasyMock.replay(mockConcurrencyMap);
+        when(mockConcurrencyMap.get(key)).thenReturn(inFlightMessage);
```

Comment:
In previous version of test there was also an expectation that mockConcurrencyMap.put() would be called. This is no longer present - is it needed?

Reply:
no we don't need that, as the production code is not calling put method on the map

Reply:
Confirmed that put() is not invoked.